src/hotspot/share/opto/c2_globals.hpp

```diff
@@ -771,6 +771,12 @@
                                                                             \
   product(bool, VerifyReceiverTypes, trueInDebug, DIAGNOSTIC,               \
           "Verify receiver types at runtime")                               \
+                                                                            \
+  product(intx, TypeProfileSubTypeCheckCommonThreshold, 50,                 \
```

Comment:
Is it better to declare it diagnostic?

Reply:
I don't have a strong opinion but the similar `TypeProfileMajorReceiverPercent` is product.